Source/WebCore/dom/FragmentDirectiveGenerator.cpp

```diff
@@ -49,12 +51,17 @@ FragmentDirectiveGenerator::FragmentDirectiveGenerator(const SimpleRange& textFr
     generateFragmentDirective(textFragmentRange);
 }
 
-static String previousWordsFromPosition(unsigned numberOfWords, VisiblePosition& startPosition)
+static bool positionsHaveSameBlockAncestor(const VisiblePosition& a, const VisiblePosition& b)
+{
+    return nearestBlockAncestor(*a.deepEquivalent().containerNode()) == nearestBlockAncestor(*b.deepEquivalent().containerNode());
```

Comment:
why is it safe to dereference these without null checks?

Reply:
fair!